### PR TITLE
[rework] ExternalXML in RAVEN Code Interface

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Same as #594 but cherry-picked for the rework branch.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
This needs to rerun tests after #595 is merged.

**Comment by alfoa** [comment]
But this one is going to create conflicts.
Hasn't be merged in devel? and consequentially in fix_conflicts branch?

**Comment by PaulTalbot-INL** [comment]
That's probably true.  If so, additional changes will be required in that merge request to get rrr-rom to pass tests, mostly changing the ext_dataobjects.xml file.

**Comment by PaulTalbot-INL** [comment]
@alfoa This can be reviewed now if you'd like, or we can wait for the Windows/Mac test machines.

**Comment by alfoa** [comment]
is this going to create conflicts ? Just to be sure.
Thnaks

File: tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/ext_dataobjects.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		<Input>DeltaTimeScramToAux,DG1recoveryTime</Input>
+		<Output>OutputPlaceHolder</Output>
+	</PointSet>
+	<PointSet name="data_for_sampling_empty_at_begin_nd">
+		<Input>DeltaTimeScramToAux,DG1recoveryTime</Input>
+		<Output>OutputPlaceHolder</Output>
+	</PointSet>
+	<PointSet name="outputMontecarloRom">
+		<Input>DeltaTimeScramToAux,DG1recoveryTime</Input>
+		<!--Output>CladTempThreshold</Output-->
```
Comment (alfoa) [inline tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/ext_dataobjects.xml]: I think this is going to create a "validation problem" of the XSD...

Comment (alfoa) [reply tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/ext_dataobjects.xml]: I do not remember if we changed the XSD schema for the DataObjects in order to accept optional <Input> and <Output>

Comment (PaulTalbot-INL) [reply tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/ext_dataobjects.xml]: It passed validation on my machine and on the qsub tests, so it seems okay.

The XSD shows minOccurs=0 for both Input and Output nodes.

Comment (alfoa) [reply tests/framework/CodeInterfaceTests/raven_running_raven_internal_models/ext_dataobjects.xml]: Ok.
We modified it. I do not like it too much but I think it is a limitation of the XSD schema now...So nothing we can do :) 

Approved

**Comment by alfoa** [comment]
This PR has been reviewed.

The MAC test in hanging because the machine is down but it has been tested locally.

Windows regression test machine needs to be moved to the new VM system

PR can be accepted.

Checklist passed.

Merging...

**Pull request merged**